app/models/armor.rb

```diff
@@ -94,12 +94,14 @@ def set_canonical_armor
     self.unit_weight = canonical_armor.unit_weight
     self.weight = canonical_armor.weight
 
-    set_enchantments if persisted?
+    set_enchantments if persisted? && canonical_armor_id_changed?
   end
 
   def set_enchantments
     return if canonical_armor.enchantments.empty?
 
+    enchantables_enchantments.added_automatically.find_each(&:destroy!)
```

Comment:
Should we create a method for this since this code occurs twice (in each of these in-game models)?